### PR TITLE
Fix up man page and add test on globs

### PR DESCRIPTION
**Pull request by rhatdan**
None
#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: test/system/060-mount.bats [hunk 1]
```diff
@@ -284,6 +284,14 @@ EOF
...
     run_podman 125 run --rm --mount source=${PODMAN_TMPDIR}/v2\*,destination=/tmp/foobar, ro=false $IMAGE touch $vol2
     is "$output" "Error: invalid reference format" "Default mounts don not support globs"
+
+    mkdir $PODMAN_TMPDIR/foo1 $PODMAN_TMPDIR/foo2 $PODMAN_TMPDIR/foo3
+    touch $PODMAN_TMPDIR/foo1/bar $PODMAN_TMPDIR/foo2/bar $PODMAN_TMPDIR/foo3/bar
+    touch $PODMAN_TMPDIR/foo1/bar1 $PODMAN_TMPDIR/foo2/bar2 $PODMAN_TMPDIR/foo3/bar3
+    run_podman 125 run --rm --mount type=glob,source=${PODMAN_TMPDIR}/foo?/bar,destination=/tmp $IMAGE ls -l /tmp
+    is "$output" "Error: /tmp/bar: duplicate mount destination" "Should report conflict on destination directory"
+    run_podman run --rm --mount type=glob,source=${PODMAN_TMPDIR}/foo?/bar?,destination=/tmp,ro $IMAGE ls /tmp
+    is "$output" "bar1.*bar2.*bar3" "Should match multiple source files on sigle destination directory"
```
Comment (edsantiago) [inline test/system/060-mount.bats:294]: typo, 'single', but not worth repushing for. LGTM, thank you!

**Review by edsantiago — APPROVED** [review]

**Review by edsantiago — APPROVED** [review]
LGTM. @TomSweeneyRedHat PTAL at man page wording.

File: docs/source/markdown/options/mount.md [hunk 2]
```diff
@@ -32,7 +32,14 @@ Current supported mount TYPEs are **bind**, **devpts**, **glob**, **image**, **t
 
 	      · dst, destination, target: mount destination spec.
 
-	      Paths matching globs, are mounted on the destination directory with the identical name inside the container.
+	      When source globs are specified without the destination directory
+              the files and directories are mounted with their complete path
+	      within the container. When the destination is specified, the
+	      files and directories matching the glob on the base file name
+	      on the destination directory. The option
```
Comment (TomSweeneyRedHat) [inline docs/source/markdown/options/mount.md:39]: I think:
```suggestion
	      on the destination directory are mounted. The option
```

File: docs/source/markdown/options/mount.md [hunk 3]
```diff
@@ -32,7 +32,14 @@ Current supported mount TYPEs are **bind**, **devpts**, **glob**, **image**, **t
...
+	      `type=glob,src=/foo*,destination=/tmp/bar` tells container engines
+	      to mount host files matching /foo* to be mounted on the /tmp/bar/
```
Comment (TomSweeneyRedHat) [inline docs/source/markdown/options/mount.md:41]:
```suggestion
	      to mount host files matching /foo* to the /tmp/bar/
```

File: docs/source/markdown/options/mount.md (hunk 2 above)
Comment (rhatdan) [reply docs/source/markdown/options/mount.md:39]: What is is trying to say is if you have

/tmp/foo1 /tmp/foo2 and do

podman run --mount type=glob,source=/tmp/foo\*:destination=/dest fedora 

/tmp/foo1 will be mounted on /dest/foo1
/tmp/foo2 will be mounted on /dest/foo2

Without destination:

podman run --mount type=glob,source=/tmp/foo\* fedora 

/tmp/foo1 -> /tmp/foo1
/tmp/foo2 -> /tmp/foo2

File: docs/source/markdown/options/mount.md [hunk 4]
```diff
@@ -32,7 +32,14 @@ Current supported mount TYPEs are **bind**, **devpts**, **glob**, **image**, **t
 
 	      · dst, destination, target: mount destination spec.
 
-	      Paths matching globs, are mounted on the destination directory with the identical name inside the container.
+	      When source globs are specified without the destination directory,
```
Comment (edsantiago) [inline docs/source/markdown/options/mount.md:35]: trailing whitespace on this line

**Comment by rhatdan** [comment]
@containers/podman-maintainers PTAL

**Pull request merged**